### ISSUE TITLE
Upgrading rancher server single node install by setting mcm to false `Failed to find system chart fleet` errors are seen in the rancher logs continuously

### ISSUE THREAD

**Issue by anupama2501** [rancher/rancher#34835]
**Rancher Server Setup**
- Rancher version: `v2.6.0`
- Installation option (Docker install/Helm Chart): Docker install
   - If Helm Chart, Kubernetes Info: k3s
         Cluster Type (RKE1, RKE2, k3s, EKS, etc):
         Node Setup: 1 node 
         Version: `v1.21.3+k3s1`
- Proxy/Cert Details: `self-signed`

After upgrade, rancher server: `v2.6-head ba43472`

**Describe the bug**
After upgraded rancher server by setting mcm flag to false, errors are seen running continuously in the rancher logs.

**To Reproduce**
1. Installed rancher server via docker `docker run -d --restart=unless-stopped  -p 80:80 -p 443:443 --privileged rancher/rancher:v2.6.0 --features=multi-cluster-management=false`
2. Verified only local cluster is available and no cluster management tab to create clusters.
3. Create a workload in the local cluster and a daemon set
4. Upgrade the rancher server to v2.6-head with the commands
```
docker stop <container_id>
docker create --volumes-from <container_id> --name rancher-data rancher/rancher:v2.6.0
docker run -d --privileged --volumes-from rancher-data --restart=unless-stopped -p 80:80 -p 443:443 rancher/rancher:v2.6-head --features=multi-cluster-management=false
```

**Result**
After a few minutes rancher comes up active. the following logs are seen continuously in the rancher server logs:
```
2021/09/20 21:47:03 [ERROR] Failed to find system chart fleet will try again in 5 seconds: no chart name found
2021/09/20 21:47:08 [ERROR] available chart version (100.0.0+up0.3.6) for fleet is less than the min version (100.0.1+up0.3.7)
```

**Expected Result**
The above error logs shouldn't be seen in the rancher server as the fleet has deployed without any error.

**Screenshots**
<img width="1520" alt="Screen Shot 2021-09-20 at 2 54 55 PM" src="https://user-images.githubusercontent.com/38144301/134081111-52276920-2c57-4cd8-b448-605206007aa6.png">

**Additional context**
Ref: https://github.com/rancher/rancher/issues/34691

**Note**: 
Issue is not seen on a fresh install of v2.6-head when the flag `--features=multi-cluster-management=false`
Not seen on a fresh install of v2.6-head when the flag `--features=multi-cluster-management=false` is not set 
Not seen on an upgrade from v2.6.0 to v2.6-head when the the flag is not set

**Comment by anupama2501** [comment]
Also seen on HA rancher pods. Upgraded the rancher server via helm `helm upgrade rancher rancher-latest/rancher --version 2.6.1-rc1  --namespace cattle-system --set hostname=anu-rdnew4.qa.rancher.space  --set rancherImageTag=v2.6-head --set 'extraEnv[0].name=CATTLE_FEATURES'   --set 'extraEnv[0].value=multi-cluster-management=false'` 

Following errors are seen in the rancher pods:
```


2021/09/21 02:12:39 [ERROR] Failed to find system chart fleet will try again in 5 seconds: no chart name found
--
Mon, Sep 20 2021 7:12:39 pm | 2021/09/21 02:12:39 [ERROR] error syncing 'local': handler provisioning-cluster-create: failed to create fleet-local/local provisioning.cattle.io/v1, Kind=Cluster for provisioning-cluster-create local: admission webhook "rancherauth.cattle.io" denied the request without explanation, requeuing


```

**Comment by anupama2501** [comment]
Also seeing these error logs upon upgrading the rancher server from v2.6.0 to v2.6.1-rc2 single node docker install
```
2021/09/24 00:01:14 [ERROR] Failed to find system chart fleet-crd will try again in 5 seconds: no chart name found
2021/09/24 00:01:19 [ERROR] available chart version (100.0.0+up0.3.6) for fleet-crd is less than the min version (100.0.1+up0.3.7)
```

**Comment by kinarashah** [comment]
Confirmed with @anupama2501 that above logs are for without passing mcm flag.

**Comment by cbron** [comment]
To repro the original bug, upgrade from 2.6.0 to 2.6.1-rc3 using a single replica.

**Comment by slickwarren** [comment]
I'm still seeing this on 2.6-head (10dfee4) following the repro steps Anupama has in this ticket:
```
[ERROR] available chart version (100.0.0+up0.3.6) for fleet is less than the min version (100.0.1+up0.3.7) 
2021/09/29 19:58:31 [ERROR] Failed to find system chart fleet will try again in 5 seconds: no chart name found
2021/09/29 19:58:32 [ERROR] error syncing 'local': handler provisioning-cluster-create: failed to create fleet-local/local provisioning.cattle.io/v1, Kind=Cluster for provisioning-cluster-create local: admission webhook "rancherauth.cattle.io" denied the request without explanation, requeuing
```

**Comment by slickwarren** [comment]
I was not able to reproduce this using a newer version (v2.6.1-rc7) of rancher following the same steps from above.
